cloudigrade/util/redhatcloud/sources.py

```diff
@@ -217,46 +224,40 @@ def notify_application_availability(
         )
         return
 
-    sources_api_base_url = settings.SOURCES_API_BASE_URL
-    sources_api_external_uri = settings.SOURCES_API_EXTERNAL_URI
-
-    url = (
-        f"{sources_api_base_url}/{sources_api_external_uri}"
-        f"applications/{application_id}/"
-    )
+    sources_kafka_config = {
+        "bootstrap.servers": f"{settings.LISTENER_SERVER}:{settings.LISTENER_PORT}"
+    }
 
     payload = {
-        "availability_status": availability_status,
-        "availability_status_error": availability_status_error,
+        "resource_type": settings.SOURCES_RESOURCE_TYPE,
+        "resource_id": application_id,
+        "status": availability_status,
+        "error": availability_status_error,
     }
 
     logger.info(
         _(
-            "Setting the availability status for application "
+            "Requesting the update of the availability status for application "
             "%(application_id)s as %(status)s"
         ),
         {"application_id": application_id, "status": availability_status},
     )
 
-    headers = identity.generate_http_identity_headers(account_number, is_org_admin=True)
-    response = requests.patch(url, headers=headers, data=json.dumps(payload))
-
-    logger.debug(
-        _("Notify sources response: %(code)s - %(response)s"),
-        {"code": response.status_code, "response": response.content},
-    )
+    try:
+        kafka_producer = KafkaProducer(sources_kafka_config)
 
-    if response.status_code == http.HTTPStatus.NOT_FOUND:
-        logger.info(
-            _(
-                "Cannot update availability status, application id "
-                "%(application_id)s not found."
-            ),
-            {"application_id": application_id},
+        kafka_producer.poll(0)
```

Comment:
What is the purpose of this `poll(0)`?

Reply:
No longer necessary, had it in the early prototype sample code.  Only useful if we register a callback. The flush we do a couple of lines later is all we need.  will remove.